org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -173,17 +173,23 @@ protected void importProjectsFromConfigurationFiles(Collection<IPath> rootPaths,
 		MultiStatus importStatusCollection = new MultiStatus(IConstants.PLUGIN_ID, -1, "Failed to import projects", null);
 		for (IPath rootPath : rootPaths) {
 			File rootFolder = rootPath.toFile();
-			try {
-				for (IProjectImporter importer : importers()) {
-					importer.initialize(rootFolder);
-					if (importer.applies(projectConfigurations, subMonitor.split(1))) {
-						importer.importToWorkspace(subMonitor.split(70));
+			Map<String, List<IPath>> configurationsByFileName = projectConfigurations.stream()
+					.filter(rootPath::isPrefixOf)
+					.collect(Collectors.groupingBy(IPath::lastSegment));
```

Comment:
Can you share a scenario (e.g. the file structure for this use case) to explain how this works?

Reply:
The previous implementation will have issue when two importer applies to the same set of configuration files. For example, Gradle build server and Buildship, both of them will import the `.gradle` files.

Now, the configuration files will be grouped according to the file name, then after any applied importer finishes the task, the loop breaks and let the next set of configuration files to be imported.

Reply:
Since there are multiple kinds of build files for gradle (e.g. build.gradle(.kts) and settings.gradle(.kts)). Group configuration by file name looks a little weird to me. For the use case below, it will get two groups, one for settings.gradle, another for build.gradle. That results in importing twice a root.

Grouping configuration by file name seems a bit odd to me, given that there are multiple types of build files for Gradle (e.g., build.gradle(.kts) and settings.gradle(.kts)). For the use case described below, two groups will be created: one for settings.gradle and another for build.gradle. This results in the root "initial" being imported twice.

```
initial
├───settings.gradle
├───application
│   ├───build.gradle
├───gradle
│   └───wrapper
└───library
    ├───settings.gradle
```

Can we reuse `importer.isResolved(...)` to break those imported configurations? Just like what `importProjects(Collection<IPath> rootPaths, IProgressMonitor monitor)` does.
https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/2412ed9d304b6458f50b512d5b0fb6adaf6fd556/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java#L151-L157